api/src/main/java/org/apache/apex/api/EmbeddedAppLauncher.java

```diff
@@ -65,6 +66,41 @@ public static EmbeddedAppLauncher newInstance()
   public interface EmbeddedAppHandle extends AppHandle {}
 
   /**
+   * <p>
+   * getDAG.</p>
+   *
+   * @return
+   */
+  public abstract DAG getDAG();
+
+  /**
+   * <p>
+   * cloneDAG.</p>
+   *
+   * @return
+   * @throws java.lang.Exception
+   */
+  public abstract DAG cloneDAG() throws Exception;
```

Comment:
These are existing methods, they cannot be changed. I just moved them up in the class hierarchy.